### DISCUSSION TITLE
Slow perfomance after update from 13.0.6 to 14.0.29

### DISCUSSION THREAD

**Discussion by snap87** [infinispan/infinispan#12674] [General]
After updating from version 13 to the latest version 14, the cache is slow. The pages began to open for the first time in a few seconds (about 10). Then they open normally. On version 13 there are no such problems and everything opens immediately. What could have influenced this?

**Comment by jabolina** [comment]
Hello, @snap87, thanks for reporting. By "the pages" do you mean the console in the browser?

**Reply by snap87** [reply]
No, by pages in this case I meant the speed of web applications.
Infinispan and our applications live in Kubernetes, this is basically just an image of a container. When changing version from 13 to 14, their work slows down significantly when first launched, but later it becomes a little faster. On the 13th version everything flies.

**Reply by jabolina** [reply]
Can you share your cache and client configuration? I assume you're running with Infinispan remotely. Is that correct? And, if you have a rough idea of which kind of operations the applications perform (gets/puts/deletes...), I might be able to run a benchmark to compare the versions.

**Reply by snap87** [reply]
```
<?xml version="1.0"?>
<infinispan xmlns="urn:infinispan:config:14.0">
    <cache-container>
        <caches>
            <distributed-cache name="ext-mem-bodies" mode="SYNC" remote-timeout="17500" statistics="true">
                <encoding>
                    <key media-type="application/json"/>
                    <value media-type="application/json"/>
                </encoding>
                <locking concurrency-level="1000" acquire-timeout="15000" striping="false"/>
                <expiration lifespan="1209600000"/>
                <state-transfer timeout="60000"/>
            </distributed-cache>
            <distributed-cache name="temp-storage" mode="SYNC" remote-timeout="17500" statistics="true">
                <encoding>
                    <key media-type="application/json"/>
                    <value media-type="application/json"/>
                </encoding>
                <locking concurrency-level="1000" acquire-timeout="15000" striping="false"/>
                <expiration lifespan="86400000"/>
                <state-transfer timeout="60000"/>
            </distributed-cache>
            <distributed-cache name="session-storage" mode="SYNC" remote-timeout="17500" statistics="true">
                <encoding>
                    <key media-type="application/json"/>
                    <value media-type="application/json"/>
                </encoding>
                <locking concurrency-level="1000" acquire-timeout="15000" striping="false"/>
                <expiration lifespan="86400000"/>
                <state-transfer timeout="60000"/>
            </distributed-cache>
            <distributed-cache name="ext-mem-indices" mode="SYNC" remote-timeout="17500" statistics="true">
                <encoding>
                    <key media-type="application/json"/>
                    <value media-type="application/json"/>
                </encoding>
                <locking concurrency-level="1000" acquire-timeout="15000" striping="false"/>
                <expiration lifespan="1209600000"/>
                <state-transfer timeout="60000"/>
            </distributed-cache>
        </caches>
    </cache-container>
</infinispan>
```
In the source code, I see that the client simply connects via hotrod to the cluster (there seem to be no other settings). hotrod://infinispan(1..3}. The application do many compute() operations.
All that is known at the moment.

**Reply by jabolina** [reply]
Thanks for sharing it. I'll find time to run our suite and see if something pops up.

I understand this is just a direct comparison between versions with the same configuration, but we can make some suggestions already. For example, `application/json` isn't the most performance-focused encoding, you could utilize Protostream instead. On the client, ensure the client intelligence is `HASH_DISTRIBUTION_AWARE`. These are just suggestions, I'll still check for regression between versions with the above configurations.

**Reply by snap87** [reply]
The problem is that I don’t see any special settings on the client. Just the address where to connect. In this case, should the client use the default setting? (she seems to be like HASH_DISTRIBUTION_AWARE).

**Reply by wburns** [reply]
Unfortunately, there isn't much to go on here. Some initial troubleshooting will help diagnose issues. I would recommend enabling access logs and you can compare the numbers between servers. https://infinispan.org/docs/stable/titles/server/server.html#enabling-access-logs_server-logging. With that you can tell immediately if it is a server issue or not and what commands would be the culprit if any.
